### ISSUE TITLE
TypeError with latest 30.0.0 version

### ISSUE THREAD

**Issue by tdruez** [nexB/scancode-toolkit#2713]
```
scancode-toolkit==30.0.0
commoncode==21.8.31
```

```
bin/scancode --info --json-pp - etc
```

```
Setup plugins...
Collect file inventory...
Scan files for: info with 1 process(es)...
Removing temporary files...done.
Traceback (most recent call last):
  File "bin/scancode", line 8, in <module>
    sys.exit(scancode())
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/commoncode/cliutils.py", line 75, in main
    **extra,
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/scancode/cli.py", line 473, in scancode
    **kwargs
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/scancode/cli.py", line 897, in run_scan
    echo_func=echo_func,
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/scancode/cli.py", line 1127, in run_scanners
    with_timing=timing, progress_manager=progress_manager)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/scancode/cli.py", line 1224, in scan_codebase
    scans = progress_manager(scans)
  File "/Volumes/nexB/repos/scancode.io/lib/python3.6/site-packages/commoncode/cliutils.py", line 285, in progressmanager
    width=width,
TypeError: __init__() got an unexpected keyword argument 'update_min_steps'
```

**Comment by tdruez** [comment]
This looks like a kwargs API issue between `commoncode==21.8.31` and `click==7.1.2`.
The `click._termui_impl.ProgressBar` has no `update_min_steps` kwarg.

**Comment by tdruez** [comment]
Nevermind, this is caused by a conflict with celery 5.1.2 that depends on click<8.0 and >=7.0 forcing the installation of a 7.0.x click that is not supported by the toolkit.
The toolkit runs fine with click > 8.0.
Closing.

**Comment by pombredanne** [comment]
> the installation of a 7.0.x click that is not supported by the toolkit.

IMHO this should work. 7.0.x is a problematic buggy click version. But 7.1 works fine with ScanCode TK?

**Comment by tdruez** [comment]
@pombredanne No sign of `update_min_steps` in https://github.com/pallets/click/blob/7.x/src/click/_termui_impl.py

**Comment by tdruez** [comment]
The `update_min_steps` kwarg was added in https://github.com/pallets/click/commit/acc91bc4f47e38f43277fcdfd8ca855734c4fbbc so starting 8.0.1
You may want to update the toolkit requirements, currently `click >= 6.7, !=7.0`

See also https://github.com/nexB/commoncode/commit/ef5f0a62f0405b9e8e2bece71d6cfe1fe19063a1#diff-0f08a9e5de99e9a72325fb0133b61561966562f5cfd23217cd187d110f3506b0L208
and https://github.com/nexB/commoncode/blob/main/setup.cfg#L40

**Comment by pombredanne** [comment]
Actually I have a patch for this at https://github.com/nexB/commoncode/issues/31

**Comment by pombredanne** [comment]
@tdruez I pushed https://github.com/nexB/scancode-toolkit/pull/2715 and your review is welcomed!

**Comment by pombredanne** [comment]
This is now fixed and released in 30.0.1
@tdruez thank you ++

**Comment by tdruez** [comment]
@pombredanne fix confirmed, thanks++
